### PR TITLE
docs: add attachment usage for mailgun

### PR DESCRIPTION
**Pull request by xixas**
### Pre-requisites

- [x] I have gone through the Contributing guidelines for [Submitting a Pull Request (PR)](../../CONTRIBUTING.md#submitting-a-pull-request-pr) and ensured that this is not a duplicate PR.
- [ ] I have performed preliminary testing using the [test suite](../../apps/api/OsmoX.postman_collection.json) to ensure that any existing features are not impacted and any new features are working as expected.
- [x] I have updated the required [api docs](../../apps/api/docs/) as applicable.
- [ ] I have added/updated test cases to the [test suite](../../apps/api/OsmoX.postman_collection.json) as applicable

### PR Details

PR details have been updated as per the given format (see below)

- [x] PR title adheres to the format specified in guidelines (e.g., `feat: add admin login endpoint`)
- [x] Description has been added
- [ ] Related changes have been added (optional)
- [ ] Screenshots have been added (optional)
- [ ] Query request and response examples have been added (as applicable, in case added or updated)
- [ ] Documentation changes have been listed (as applicable)
- [ ] Test suite output is added (as applicable)
- [ ] Pending actions have been added (optional)
- [ ] Any other additional notes have been added (optional)

### Additional Information

- [x] Appropriate label(s) have been added (`ready for review` should be added if the PR is ready to be reviewed)
- [x] Assignee(s) and reviewer(s) have been added (optional)

---

**Description:**

Updated document with attachment usage.
Fixed type issues using ChatGPT

**Related changes:**

NA

**Screenshots:**

NA

**Query request and response:**

NA

**Documentation changes:**

NA

**Test suite output:**

NA

**Pending actions:**

NA

**Additional notes:**

NA

## Summary by CodeRabbit

- **Documentation**
  - Enhanced Mailgun API documentation with clearer instructions for configuration updates.
  - Added guidance on handling attachments with multiple MIME types using base64 encoding in Node.js and .NET.
  - Included sample JSON configurations and code snippets for creating base64 strings in Node.js and .NET.
  - Improved `Dependencies` section formatting.

### CODE REVIEW

File: apps/api/docs/channels/mailgun.md [hunk 1]
```diff
@@ -4,26 +4,26 @@ Mailgun is an email service provider that offers a RESTful API for sending email
...
 
 | Key             | Description                                                     |
 |-----------------|-----------------------------------------------------------------|
 | MAILGUN_API_KEY | Your Mailgun API key                                            |
 | MAILGUN_HOST    | Mailgun host; api.mailgun.net for US, api.eu.mailgun.net for EU |
 | MAILGUN_DOMAIN  | Your Mailgun domain name                                        |
 
-```jsonc
-// Sample json to set in configuration field
+```json
```
Comment (LakshayaT) [inline apps/api/docs/channels/mailgun.md:22]: Can we continue to use `jsonc` here as without it, comments get highlighted in red as invalid code in readme:
image

File: apps/api/docs/channels/mailgun.md [hunk 2]
```diff
@@ -34,7 +34,7 @@ Make sure to replace `your-api-key`, `api.mailgun.net`, and `your.mailgun.domain
 
 Here's a sample request body:
 
-```jsonc
+```json
```
Comment (LakshayaT) [inline apps/api/docs/channels/mailgun.md:37]: Same here, can we use `jsonc` only?

Comment (xixas) [reply apps/api/docs/channels/mailgun.md]: Updated

File: apps/api/docs/channels/mailgun.md (hunk 1 above)
Comment (xixas) [reply apps/api/docs/channels/mailgun.md]: Reverted Back

**Comment by LakshayaT** [comment]
@xixas I also think that we pass attachments in the query as follows:

```jsonc
    "attachments": [                               // Attachments (optional)
      {
        "filename": "Somefile.type",
        "content": "Somecontentstring",
      },
    ],
```

instead of 

```jsonc
    "attachment": [                                 // Attachments (optional)
      {
        "filename": "Somefile.type",
        "data": "Somecontentstring",
      },
    ],
```

Can you please cross check this with the request you sent for testing and update in the document as per that?

File: apps/api/docs/channels/mailgun.md [hunk 3]
```diff
@@ -61,15 +61,46 @@ In addition to the provided fields, there are several other options you can incl
...
-| --------------- | ------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------- |
-| mailgun.js       | ^10.2.1   | A Mailgun client for JavaScript, used to interact with the Mailgun API.     |
-| form-data           | ^4.0.0 | Used by mailgun.js to handle form data for HTTP requests.     | 
+| Package Name  | Version | Description                                     |
+|---------------|---------|-------------------------------------------------|
+| mailgun.js    | ^10.2.1 | A Mailgun client for JavaScript                 |
+| form-data     | ^4.0.0  | Used by mailgun.js to handle form data for HTTP requests |
 
-Reference: https://documentation.mailgun.com/en/latest/api-sending.html
+Reference: https://documentation.mailgun.com/en/latest/api-sending.html
```
[the remark this answers is not in the document]
Comment (xixas) [reply apps/api/docs/channels/mailgun.md:106]: Updated

**Comment by xixas** [comment]
> @xixas I also think that we pass attachments in the query as follows:
> 
> ```js
>     "attachments": [                               // Attachments (optional)
>       {
>         "filename": "Somefile.type",
>         "content": "Somecontentstring",
>       },
>     ],
> ```
> 
> instead of
> 
> ```js
>     "attachment": [                                 // Attachments (optional)
>       {
>         "filename": "Somefile.type",
>         "data": "Somecontentstring",
>       },
>     ],
> ```
> 
> Can you please cross check this with the request you sent for testing and update in the document as per that?

Yes, Its content in API Call.

**Review by LakshayaT — APPROVED** [review]

**Pull request merged**